channel-azure-boards/src/main/java/com/synopsys/integration/alert/channel/azure/boards/oauth/storage/AzureBoardsCredentialDataStore.java

```diff
@@ -47,48 +49,94 @@ protected AzureBoardsCredentialDataStore(AzureBoardsCredentialDataStoreFactory d
     public Collection<StoredCredential> values() throws IOException {
-        ConfigurationModel configurationModel = retrieveConfiguration();
-        StoredCredential storedCredential = createStoredCredential(configurationModel);
-        return Set.of(storedCredential);
+        lock.lock();
+        try {
+            ConfigurationModel configurationModel = retrieveConfiguration();
+            StoredCredential storedCredential = createStoredCredential(configurationModel);
+            return Set.of(storedCredential);
+        } finally {
+            lock.unlock();
+        }
     }
 
     @Override
     public StoredCredential get(String key) throws IOException {
-        if (null == key) {
-            return null;
+        lock.lock();
+        try {
+            if (null == key) {
+                return null;
+            }
+            return retrieveCredentialMatchingEmailOrNull(key);
+        } finally {
+            lock.unlock();
         }
-        return retrieveCredentialMatchingEmailOrNull(key);
     }
 
     @Override
     public AzureBoardsCredentialDataStore set(String key, StoredCredential value) throws IOException {
-        if (null != key && null != value) {
-            ConfigurationModel defaultConfig = retrieveConfiguration();
-            Map<String, ConfigurationFieldModel> keyToFieldMap = defaultConfig.getCopyOfKeyToFieldMap();
-            setFieldValue(keyToFieldMap, AzureBoardsDescriptor.KEY_OAUTH_USER_EMAIL, key);
-            setFieldValue(keyToFieldMap, AzureBoardsDescriptor.KEY_ACCESS_TOKEN, value.getAccessToken());
-            setFieldValue(keyToFieldMap, AzureBoardsDescriptor.KEY_REFRESH_TOKEN, value.getRefreshToken());
-
-            Long expTimeMillis = value.getExpirationTimeMilliseconds();
-            String expTimeMillisString = expTimeMillis != null ? expTimeMillis.toString() : null;
-            setFieldValue(keyToFieldMap, AzureBoardsDescriptor.KEY_TOKEN_EXPIRATION_MILLIS, expTimeMillisString);
-
-            try {
-                configurationModelConfigurationAccessor.updateConfiguration(defaultConfig.getConfigurationId(), keyToFieldMap.values());
-            } catch (AlertConfigurationException e) {
-                throw new IOException("Cannot update the Azure Boards global configuration", e);
+        lock.lock();
+        try {
+            if (null != key && null != value) {
+                boolean changed;
+                ConfigurationModel defaultConfig = retrieveConfiguration();
+                Map<String, ConfigurationFieldModel> keyToFieldMap = defaultConfig.getCopyOfKeyToFieldMap();
+                changed = isFieldValueChanged(keyToFieldMap, AzureBoardsDescriptor.KEY_OAUTH_USER_EMAIL, key);
+                changed = changed || isFieldValueChanged(keyToFieldMap, AzureBoardsDescriptor.KEY_ACCESS_TOKEN, value.getAccessToken());
+                changed = changed || isFieldValueChanged(keyToFieldMap, AzureBoardsDescriptor.KEY_REFRESH_TOKEN, value.getRefreshToken());
```

Comment:
Should this be one long logical expression rather than broken up, or was this for readability?

Reply:
For readability and in case we need to debug an issue you know which field has changed.